js-api-doc/value/calculation.d.ts

```diff
@@ -0,0 +1,141 @@
+import {List, ValueObject} from 'immutable';
+import {Value, SassNumber, SassString} from './index';
+
+/** The type of values that can be arguments to a SassCalculation. */
+type CalculationValue =
+  | SassNumber
+  | SassCalculation
+  | SassString
+  | CalculationOperation
+  | CalculationInterpolation;
+
+/**
+ * Sass's [calculation
+ * type](https://sass-lang.com/documentation/values/calculations).
+ *
+ * Note: in the JS API calculations are not simplified eagerly. This also means
+ * that unsimplified calculations are not equal to the numbers they would be
+ * simplified to.
+ *
+ * @category Custom Function
+ */
+export class SassCalculation extends Value {
+  /**
+   * Creates a value that represents `calc(argument)`.
+   *
+   * @throws `Error` if `argument` is or transitively contains a quoted
+   * {@link SassString}
+   * @returns A calculation with the name `calc` and `argument` as its single
+   * argument.
+   */
+  static calc(argument: CalculationValue): SassCalculation;
+
+  /**
+   * Creates a value that represents `min(arguments...)`.
+   *
+   * @throws `Error` if any of `arguments` are or transitively contain a quoted
+   * {@link SassString}
+   * @returns A calculation with the name `min` and `arguments` as its arguments.
+   */
+  static min(
+    arguments: CalculationValue[] | List<CalculationValue>
+  ): SassCalculation;
+
+  /**
+   * Creates a value that represents `max(arguments...)`.
+   *
+   * @throws `Error` if any of `arguments` are or transitively contain a quoted
+   * {@link SassString}
+   * @returns A calculation with the name `max` and `arguments` as its arguments.
+   */
+  static max(
+    arguments: CalculationValue[] | List<CalculationValue>
+  ): SassCalculation;
+
+  /**
+   * Creates a value that represents `clamp(value, min, max)`.
+   *
+   * @throws `Error` if any of `value`, `min`, or `max` are or transitively
+   * contain a quoted {@link SassString}.
+   * @throws `Error` if `value` is undefined and `max` is not undefined.
+   * @throws `Error` if `value` or `max` is undefined and neither `min` nor
+     `value` is a {@link SassString} that begins with `var(`
```

Comment:
I think this is actually wrong in the spec. It should allow any string or interpolation to be considered as multiple arguments, because it _could_ contain a comma (e.g. `clamp(#{"1, 2, 3"})`)